packages/task/src/browser/task-service.ts

```diff
@@ -408,12 +454,184 @@ export class TaskService implements TaskConfigurationClient {
+        if (task && task.dependsOn) {
+            // In case the 'dependsOn' is an array
+            if (Array.isArray(task.dependsOn) && task.dependsOn.length > 0) {
+                for (let i = 0; i < task.dependsOn.length; i++) {
+                    const childNode = this.createChildTaskNode(task, taskNode, task.dependsOn[i], tasks);
+                    this.detectDirectedAcyclicGraph(childNode.taskConfiguration, childNode.node, tasks);
+                }
+            } else if (!Array.isArray(task.dependsOn)) {
+                const childNode = this.createChildTaskNode(task, taskNode, task.dependsOn, tasks);
+                this.detectDirectedAcyclicGraph(childNode.taskConfiguration, childNode.node, tasks);
+            }
+        }
+    }
+
+    // 'childTaskIdentifier' may be a string (a task label) or a JSON object which represents a TaskIdentifier (e.g. {"type":"npm", "script":"script1"})
+    createChildTaskNode(task: TaskConfiguration, taskNode: TaskNode, childTaskIdentifier: string | TaskIdentifier, tasks: TaskConfiguration[]): TaskGraphNode {
+        const childTaskConfiguration = this.getDependentTask(childTaskIdentifier, tasks);
+
+        // If current task and child task are identical or if
+        // one of the child tasks is identical to one of the current task ancestors, then raise an error
+        if (this.taskDefinitionRegistry.compareTasks(task, childTaskConfiguration) ||
+            taskNode.parentsID.filter(t => this.taskDefinitionRegistry.compareTasks(childTaskConfiguration, t)).length > 0) {
+            const fromNode = task.label;
+            const toNode = childTaskConfiguration.label;
+            throw new Error('Circular reference detected: ' + fromNode + ' -->  ' + toNode);
+        }
+        const childNode = new TaskNode(childTaskConfiguration, [], Object.assign([], taskNode.parentsID));
+        childNode.addParentDependency(taskNode.taskId);
+        taskNode.addChildDependency(childNode);
+        return { 'taskConfiguration': childTaskConfiguration, 'node': childNode };
+    }
+
+    /**
+     * Gets task configuration by task label or by a JSON object which represents a task identifier
+     *
+     * @param taskIdentifier The task label (string) or a JSON object which represents a TaskIdentifier (e.g. {"type":"npm", "script":"script1"})
+     * @param tasks an array of the task configurations
+     * @returns the correct TaskConfiguration object which matches the taskIdentifier
+     */
+    getDependentTask(taskIdentifier: string | TaskIdentifier, tasks: TaskConfiguration[]): TaskConfiguration {
+        const notEnoughDataError = 'The information provided in the "dependsOn" is not enough for matching the correct task !';
+        let currentTaskChildConfiguration: TaskConfiguration;
+        if (typeof (taskIdentifier) !== 'string') {
+            // TaskIdentier object does not support tasks of type 'shell' (The same behavior as in VS Code).
```

Comment:
[minor] typo - `TaskIdentier` should be `TaskIndentifier`.

Reply:
fixed, done